parlai/agents/hred/hred.py

```diff
@@ -60,7 +65,7 @@ def add_cmdline_args(cls, argparser):
             help="Probability of replacing tokens with UNK in training.",
         )
 
-        super(HredAgent, cls).add_cmdline_args(argparser)
+        super().add_cmdline_args(parser, partial_opt=partial_opt)
         return agent
```

Comment:
kind of a nit (this applies to other classes as well), but `return agent` here returns the argument group, not a `ParlaiParser` (unless my understanding of the return value of `add_argument_group` is incorrect)

Reply:
This is worth debating. We use the `add_argument_group` bit quite a bit, in order to add sub-arguments of groups based on classes.

We could ban returning argument groups: every subclass should make a new group. Not sure? It's unclear to me that the class hierarchy is what should be reflected in the config options.

Reply:
i think the argument groups improve readability tremendously, especially when using e.g. `parlai tm -m my_agent -h`, as you can easily read through args based on who added them

where do we use this returned parser anyway?

Reply:
In superclass:

group  = parser.add_argument_group()
return group

In subclass:

parser = super().add_argument_group()
parser.add_argument('--subclass-option')
return parser

Reply:
Anyway I thought about this and figured our typechecker isn't strict, let's just violate it

Reply:
if it's just a matter of violating the typechecker (and nothing functionally is incorrect) I agree

Reply:
Maybe the really clean thing would be to have the return value always be an ArgumentGroup but idk if I can handle that codemod

Reply:
you could modify the return to be `Union[ParlaiParser, ArgumentGroup]`, seems like an easier codemod

Reply:
I've opted not to do that because it just is a lot of noise. We'll put this off for future better eng.